haystack/document_stores/memory.py

```diff
@@ -113,6 +113,8 @@ def __init__(
         self.bm25_parameters = bm25_parameters
         self.bm25: Dict[str, rank_bm25.BM25] = {}
 
+        torch_import.check()
+
```

Comment:
Nitpick: I would always put these check at the very first line of the init. If the deps are not there, no point doing anything else, running any super(), making any checks, etc... It should be the first issue the users are warned about.